src/main/java/basesyntax/Main.java

```
@@ -0,0 +1,58 @@
+import basesyntax.service.shop.ShopServiceImpl;
+import basesyntax.service.writer.FileWriter;
+import basesyntax.service.writer.FileWriterImpl;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    private static final String INPUT_FILE_PATH = "src/main/resources/data.csv";
+    private static final String OUTPUT_FILE_PATH = "src/main/resources/report.csv";
+
+    public static void main(String[] args) {
+        FileReader fileReader = new FileReaderImpl();
+        DataConverter dataConverter = new DataConverterImpl();
+
+        Map<FruitTransaction.Operation, OperationHandler> handlerMap = new HashMap<>();
+        handlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceOperation());
+        handlerMap.put(FruitTransaction.Operation.SUPPLY, new SupplyOperation());
+        handlerMap.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperation());
+        handlerMap.put(FruitTransaction.Operation.RETURN, new ReturnOperation());
+
+        OperationStrategy operationStrategy = new OperationStrategyImpl(handlerMap);
+        ShopService shopService = new ShopServiceImpl(operationStrategy);
+        ReportGenerator reportGenerator = new ReportGeneratorImpl();
+        FileWriter fileWriter = new FileWriterImpl();
+
+        try {
+            List<String> rawData = fileReader.read(INPUT_FILE_PATH);
+            List<FruitTransaction> transactions = dataConverter.convertToTransaction(rawData);
+            shopService.process(transactions);
+            String report = reportGenerator.getReport();
+            fileWriter.write(report, OUTPUT_FILE_PATH);
+            System.out.println("The report is successfully generated into a file: "
+                    + OUTPUT_FILE_PATH);
+        } catch (RuntimeException | IOException e) {
+            System.err.println("An error occurred while processing data: " + e.getMessage());
+            e.printStackTrace();
```

This violates checklist item: "Don't ignore exceptions in try-catch blocks". You should not print stack traces or error messages directly in production code. Instead, rethrow as a RuntimeException with a clear message, or handle as per your application's error policy.